modules/auxiliary/client/telegram/send_message.rb

```diff
@@ -3,54 +3,104 @@
-            on how to retrieve the bot token and corresponding chat ID values.
+        This module can be used to send a specified document and message
+        to multiple users with an optional message and can be used for
+        ethical phishing campaigns. Please refer to the module documentation
+        for info on how to retrieve the bot token and corresponding chat ID
+        values.
         },
-    'Author' => 'Ege Balcı <egebalci[at]pm.me>', # Aka @egeblc of https://pentest.blog
+    'Author' =>
+      [
+        'Ege Balcı <egebalci[at]pm.me>', # Aka @egeblc of https://pentest.blog
+        'Gaurav Purswani' # @pingport80
+      ],
     'License' => MSF_LICENSE,
     )
 
     register_options(
       [
         OptString.new('BOT_TOKEN', [true, 'Telegram BOT token', '']),
-        OptInt.new('CHAT_ID', [true, 'Chat ID for the BOT', '']),
-        OptString.new('MSG', [true, 'Message content', 'New session opened!']),
-        OptEnum.new('FORMATTING', [true, 'Message formating option (Markdown|MarkdownV2|HTML)', 'Markdown', [ 'Markdown', 'MarkdownV2', 'HTML']]),
+        OptString.new('MESSAGE', [false, 'Optional message sent with the document']),
+        OptInt.new('CHAT_ID', [false, 'Chat ID for the BOT', '']),
+        OptPath.new('DOCUMENT', [false, 'The path to the document(binary, video etc)']),
+        OptPath.new('IDFILE', [false, 'File containing chat IDs, one per line'])
       ], self.class
     )
   end
 
   def message
-    datastore['MSG']
+    datastore['MESSAGE']
   end
 
-  def formatting
-    datastore['FORMATTING']
+  def document
+    datastore['DOCUMENT']
   end
 
   def bot_token
     datastore['BOT_TOKEN']
   end
 
-  def run
-    uri = URI("https://api.telegram.org/bot#{bot_token}/sendMessage")
-    params = { chat_id: datastore['CHAT_ID'], parse_mode: formatting, text: message }
-    uri.query = URI.encode_www_form(params)
-    res = Net::HTTP.get_response(uri)
+  def id_file
+    datastore['IDFILE']
+  end
 
-    if res.is_a?(Net::HTTPSuccess)
-      print_good('Message sent!')
+  def send_document(conn, chat_id)
+    return unless document
+
+    raw_params = { 'chat_id' => chat_id, 'document' => Faraday::UploadIO.new(document, 'application/octet-stream') }
+    parms = {}
+    raw_params.each_with_object({}) do |(key, value), _params|
+      parms[key] = value
+    end
+    print_warning("Sending to #{chat_id.strip}")
+    response = conn.post("/bot#{bot_token}/sendDocument", parms)
+    if response.status == 200
+      print_good('Document sent successfully!')
+    elsif response.status == 403
+      print_bad("Error while sending document! Make sure the user with chat_id : #{chat_id} is active on bot.")
     else
-      print_error('Unable to send the message!')
-      print_error("API Status: #{res.code}")
+      print_bad('Error while sending the document!')
     end
   end
+
+  def send_message(conn, chat_id)
+    return if message == ''
+
+    parms = { 'chat_id' => chat_id, 'text' => message }
```

Comment:
See my `parse_mode` comment above, I think the formatting option should be added in here. Can understand it not being added to documents since it doesn't really affect those but the ability to send markdown and HTML are useful features that should be kept in this module.

Reply:
Oh, sorry I think I removed it by mistake, I'll add this :+1:

Reply:
yeah, I'll add this back. :+1: